### PR TITLE
added adal-library files

### PR DESCRIPTION
**Pull request by cssuh**
Didn't want the other PR to get too big so I created another one with just the adal-library files.  They've already been reviewed for the most part from #1727

### CODE REVIEW

File: lib/adal-aad-node/src/models/error.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    getSimpleError: (errorCode: ErrorCodes) => string;
+
+    getTenantNotFoundError: (context: Error1Context) => string;
+}
+
+export interface Error1Context {
+    tenantId: string;
+}
+
+
```
Comment (abist) [inline lib/adal-aad-node/src/models/error.ts:13]: nit: extra newlines

File: lib/adal-aad-node/src/engine/AzureAuth.ts [hunk 2]
```diff
@@ -0,0 +1,469 @@
+import { ProviderSettings, SecureStorageProvider, Tenant, AADResource, LoginResponse, Deferred, AzureAccount, Logger, MessageDisplayer, ErrorLookup, CachingProvider, RefreshTokenPostData, AuthorizationCodePostData, TokenPostData, AccountKey, StringLookup, DeviceCodeStartPostData, DeviceCodeCheckPostData, AzureAuthType, AccountType, UserInteraction } from '../models';
```
Comment (abist) [inline lib/adal-aad-node/src/engine/AzureAuth.ts:1]: Please add the `Microsoft` license header in all the `.ts` files you're checking in

**Review by abist — CHANGES_REQUESTED** [review]
Please add the copyright header in all of the `.ts` files you're checking in

**Comment by abist** [comment]
The code review is the same from last time, so it's fine. Please add the header copy right comments, however.

**Comment by cssuh** [comment]
yep, just added the copyright statements

**Review by abist — APPROVED** [review]

**Pull request merged**